src/main/kotlin/com/spectralogic/rioBench/commands/Benchmark.kt

```diff
@@ -0,0 +1,87 @@
+ */
+
+package com.spectralogic.rioBench.commands
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.choice
+import com.github.ajalt.clikt.parameters.types.long
+import com.spectralogic.escapepod.devintegration.FileToArchive
+import com.spectralogic.escapepod.devintegration.FilesToArchive
+import com.spectralogic.escapepod.devintegration.RioClient
+import com.spectralogic.escapepod.devintegration.UserLoginCredentials
+import com.spectralogic.escapepod.devintegration.createRioAuthClient
+import com.spectralogic.escapepod.devintegration.createRioClient
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.joinAll
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import java.net.URI
+import java.time.Instant
+import kotlin.math.pow
+import kotlin.math.roundToLong
+
+class Benchmark : CliktCommand(help = "Run the benchmark suite", name = "benchmark") {
+    private val host: String by option("-h", "--hostname", help = "Rio hostname").required()
+    private val broker: String by option("-b", "--broker", help = "Name of the broker to use").required()
+    private val jobs: Long by option("-j", "--jobs", help = "number of jobs to launch").long().required()
+    private val number: Long by option("-n", "--number", help = "number of items per job").long().required()
+    private val size: Long by option("-s", "--size", help = "size of an individual item").long().required()
+    private val units: Int by option("-u", "--units", help = "units for job item")
+        .choice(Pair("b", 0), Pair("kb", 1), Pair("Mb", 2), Pair("Gb", 3), Pair("Tb", 4))
+        .required()
+
+    override fun run() {
+        runBlocking {
+            val timedRioClient = TimedRioClient(host, UserLoginCredentials("spectra", "spectra"))
+            val itemSize: Long = size * (1024.0.pow(units)).roundToLong()
+            (0 until jobs).map { job ->
+                (0 until number).map { file ->
+                    FileToArchive(
+                        "$job-$file-${Instant.now().toEpochMilli()}",
+                        URI.create("aToZSequence://f"),
+                        itemSize
+                    )
+                }.toList()
+            }
+                .map { fileToArchive ->
+                    GlobalScope.launch (IO) {
+                        timedRioClient.rioClient.archiveFiles(broker, FilesToArchive(fileToArchive))
+                    }
+                }
+                .joinAll()
+        }
+        echo("Finished sending jobs")
+    }
+
+    inner class TimedRioClient(private val host: String, private val credentials: UserLoginCredentials) {
+        private var lastCreated: Instant = Instant.now()
+        var rioClient: RioClient =
```

Comment:
I thought we had modified this so that it wasn't a property and was suspending?

Reply:
We did, my bad.